### PR TITLE
V2:Astro-913 - Segmented Button

### PR DESCRIPTION
**Pull request by im-cr**
**Ticket**
- 913: https://rocketcom.atlassian.net/browse/ASTRO-913

**Changes** - commit: https://github.com/RocketCommunicationsInc/astro-components/commit/4eaad9100d584064f1abed0b35b2a7dd4271f4ad
 - theme-dark
 - theme-light
 - segmented-button.css
 - rux-segmented-button.js

**Design References**
  - [Dark Colors](https://app.abstract.com/projects/2d8cf270-0b93-11e8-89ff-61554354acb1/branches/f22ccbcd-6954-4228-b52f-34e5288e75c8/files/1d17329d-e1a3-490a-a6ad-097e2ecd4825?filter=changed)
  - [Light Colors](https://app.abstract.com/projects/2d8cf270-0b93-11e8-89ff-61554354acb1/branches/f22ccbcd-6954-4228-b52f-34e5288e75c8/files/b7c2bf0c-9dec-4b07-96d4-ead85393c9d0?filter=changed&s=segmented)

### CODE REVIEW

File: src/css/common/_theme--dark.css [hunk 1]
```diff
@@ -2,28 +2,42 @@
...
-  --fontLowContrastColor: var(--colorGray, rgb(204, 204, 204));
-  --fontInvertedColor: var(--colorBlack, rgb(0, 0, 0));
-  --fontInvertedBackgroundColor: var(--colorWhite, rgb(255, 255, 255));
-
-  --fontLinkColor: var(--colorSecondary, rgb(77, 172, 255));
-  --fontLinkHoverColor: var(--colorSecondaryLighten1, rgb(113, 189, 255));
+    =========================================================================
+	*/
+
+	/* KM Dark Palette Colors */
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:8]: Change to Astro 5 Simplified Colors

File: src/css/common/_theme--dark.css [hunk 2]
```diff
@@ -112,57 +115,49 @@
...
   /* 
     
     Segmented Button Colors 
     ==========================================================================
 
   */
-  --segmentedButtonBorderColor: var(--colorTertiaryDarken1, rgb(30, 47, 66));
-  --segmentedButtonBackgroundColor: var(--colorPrimary, rgb(0, 90, 143));
-  --segmentedButtonTextColor: var(--colorWhite, rgb(255, 255, 255));
+  --segmentedButtonBorderColor: var(--colorTertiaryDarken1);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:134]: should be --primary

File: src/css/common/_theme--dark.css [hunk 3]
```diff
@@ -112,57 +115,49 @@
...
+  --segmentedButtonBackgroundColor: none;
+  --segmentedButtonTextColor: var(--primaryLight);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:136]: Should be --pimary

File: src/css/common/_theme--light.css [hunk 4]
```diff
@@ -3,26 +3,40 @@
...
-  --fontColor: var(--colorBlack, rgb(0, 0, 0));
-  --fontLowContrastColor: var(--colorQuaternaryDarken1, rgb(166, 171, 182));
-  --fontInvertedColor: var(--colorWhite, rgb(255, 255, 255));
-  --fontInvertedBackgroundColor: var(--colorBlack, rgb(0, 0, 0));
-
-  --fontLinkColor: var(--colorPrimary, rgb(0, 90, 143));
-  --fontLinkHoverColor: var(--colorSecondaryDarken1, rgb(58, 129, 191));
+	*/
+	
+	/* KM Light Palette Colors */
```
Comment (dmcalester) [inline src/css/common/_theme--light.css:8]: Change to Astro 5 Simplified Colors

File: src/css/common/_theme--light.css [hunk 5]
```diff
@@ -73,116 +84,95 @@
...
+  --segmentedButtonBorderColor: var(--primary);
+  --segmentedButtonBackgroundColor: none;
+  --segmentedButtonTextColor: var(--primary);
 
-  --segmentedButtonSelectedBackgroundColor: var(
-    --colorSecondaryDarken1,
-    rgb(58, 129, 191)
-  );
+	/* Segmented Hover */
+  --segmentedButtonHoverBackgroundColor: var(--primaryLight);
```
Comment (dmcalester) [inline src/css/common/_theme--light.css:156]: should be `--primaryDark`

File: src/css/common/_theme--light.css [hunk 6]
```diff
@@ -73,116 +84,95 @@
...
+  --segmentedButtonHoverTextColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--light.css:157]: should be `--defaultText`

File: src/css/common/_theme--light.css [hunk 7]
```diff
@@ -73,116 +84,95 @@
...
-  --segmentedButtonSelectedBoxShadow: inset 0 3px 4px rgba(0, 0, 0, 0.1),
-    inset 0 1px 5px rgba(0, 0, 0, 0.06);
-  transition: box-shadow 0.067s ease-out;
+	/* Segmented Select */
+  --segmentedButtonSelectedBackgroundColor: var(--primary);
+	--segmentedButtonSelectedBoxShadow: inset 0 3px 4px rgba(0, 0, 0, 0.1), inset 0 1px 5px rgba(0, 0, 0, 0.06);
+  	transition: box-shadow 0.067s ease-out;
 
-  --segmentedButtonSelectedTextColor: var(--colorWhite, rgb(255, 255, 255));
+  --segmentedButtonSelectedTextColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--light.css:164]: should be `--defaultText`

File: src/css/common/_theme--light.css (hunk 6 above)
Comment (dmcalester) [inline src/css/common/_theme--light.css:157]: should be `--primaryElementText`

File: src/css/common/_theme--light.css (hunk 7 above)
Comment (dmcalester) [inline src/css/common/_theme--light.css:164]: should be `--primaryElementText`

File: static/css/astro.css [hunk 8]
```diff
@@ -3374,36 +3370,15 @@ i.rux-icon {
...
-  box-shadow: var(
-    --segmentedButtonSelectedBoxShadow,
-    inset 0 2px 4px rgba(0, 0, 0, 0.14),
-    inset 0 3px 4px rgba(0, 0, 0, 0.12),
-    inset 0 1px 5px rgba(0, 0, 0, 0.2)
-  );
+  background-color: var(--segmentedButtonSelectedBackgroundColor);
+  color: var(--segmentedButtonSelectedTextColor);
+  box-shadow: var(--segmentedButtonSelectedBoxShadow);
   transition: box-shadow 0.067 ease-out;
```
Comment (dmcalester) [inline static/css/astro.css:3382]: remove transition

File: static/css/astro.css [hunk 9]
```diff
@@ -3374,36 +3370,15 @@ i.rux-icon {
...
-  ;
-  box-shadow: var(
-    --segmentedButtonSelectedBoxShadow,
-    inset 0 2px 4px rgba(0, 0, 0, 0.14),
-    inset 0 3px 4px rgba(0, 0, 0, 0.12),
-    inset 0 1px 5px rgba(0, 0, 0, 0.2)
-  );
+  background-color: var(--segmentedButtonSelectedBackgroundColor);
+  color: var(--segmentedButtonSelectedTextColor);
+  box-shadow: var(--segmentedButtonSelectedBoxShadow);
```
Comment (dmcalester) [inline static/css/astro.css:3381]: remove box-shadow

File: static/css/astro.css [hunk 10]
```diff
@@ -510,19 +510,19 @@
...
-  --segmentedButtonSelectedBackgroundColor: var(--colorSecondaryDarken1);
-
-  --segmentedButtonSelectedBoxShadow: inset 0 3px 4px rgba(0, 0, 0, 0.1),
-    inset 0 1px 5px rgba(0, 0, 0, 0.06);
-  transition: box-shadow 0.067s ease-out;
-
-  --segmentedButtonSelectedTextColor: var(--colorWhite);
+	/* Segmented Select */
+  --segmentedButtonSelectedBackgroundColor: var(--primary);
+	/* --segmentedButtonSelectedBoxShadow: inset 0 3px 4px rgba(0, 0, 0, 0.1), inset 0 1px 5px rgba(0, 0, 0, 0.06);
```
Comment (dmcalester) [inline static/css/astro.css:522]: remove all references to box shadow

File: static/css/astro.css [hunk 11]
```diff
@@ -924,22 +924,23 @@ li.light-theme {
...
+  --segmentedButtonHoverBackgroundColor: var(--primaryLight);
+  --segmentedButtonHoverTextColor: var(--primaryElementText);
 
-  --segmentedButtonSelectedBoxShadow: inset 0 2px 4px rgba(0, 0, 0, 0.14),
-    inset 0 3px 4px rgba(0, 0, 0, 0.12), inset 0 1px 5px rgba(0, 0, 0, 0.2);
+	/* Segmented Select */
+  --segmentedButtonSelectedBackgroundColor: var(--primary);
 
-  --segmentedButtonSelectedTextColor: var(--colorWhite);
+	/* --segmentedButtonSelectedBoxShadow:
```
Comment (dmcalester) [inline static/css/astro.css:938]: remove all referenced to box-shadow

File: static/css/astro.core.css [hunk 12]
```diff
@@ -924,22 +924,23 @@ li.light-theme {
...
+  --segmentedButtonHoverBackgroundColor: var(--primaryLight);
+  --segmentedButtonHoverTextColor: var(--primaryElementText);
 
-  --segmentedButtonSelectedBoxShadow: inset 0 2px 4px rgba(0, 0, 0, 0.14),
-    inset 0 3px 4px rgba(0, 0, 0, 0.12), inset 0 1px 5px rgba(0, 0, 0, 0.2);
+	/* Segmented Select */
+  --segmentedButtonSelectedBackgroundColor: var(--primary);
 
-  --segmentedButtonSelectedTextColor: var(--colorWhite);
+	/* --segmentedButtonSelectedBoxShadow:
```
Comment (dmcalester) [inline static/css/astro.core.css:938]: remove all references to box shadow

File: src/css/components/segmented-button.css [hunk 13]
```diff
@@ -60,25 +60,14 @@
...
-  color: var(--segmentedButtonSelectedTextColor, rgb(255, 255, 255));
-  box-shadow: var(
-    --segmentedButtonSelectedBoxShadow,
-    inset 0 2px 4px rgba(0, 0, 0, 0.14),
-    inset 0 3px 4px rgba(0, 0, 0, 0.12),
-    inset 0 1px 5px rgba(0, 0, 0, 0.2)
-  );
+  background-color: var(--segmentedButtonSelectedBackgroundColor);
+  color: var(--segmentedButtonSelectedTextColor);
+  box-shadow: var(--segmentedButtonSelectedBoxShadow);
```
Comment (dmcalester) [inline src/css/components/segmented-button.css:71]: Remove all box-shadow props

File: src/css/components/segmented-button.css [hunk 14]
```diff
@@ -60,25 +60,14 @@
...
   transition: box-shadow 0.067 ease-out;
```
Comment (dmcalester) [inline src/css/components/segmented-button.css:72]: Remove transition

File: src/css/common/_theme--light.css [hunk 15]
```diff
@@ -149,19 +149,19 @@
...
-  --segmentedButtonSelectedBackgroundColor: var(--colorSecondaryDarken1);
-
-  --segmentedButtonSelectedBoxShadow: inset 0 3px 4px rgba(0, 0, 0, 0.1),
-    inset 0 1px 5px rgba(0, 0, 0, 0.06);
-  transition: box-shadow 0.067s ease-out;
-
-  --segmentedButtonSelectedTextColor: var(--colorWhite);
+	/* Segmented Select */
+  --segmentedButtonSelectedBackgroundColor: var(--primary);
+	/* --segmentedButtonSelectedBoxShadow: inset 0 3px 4px rgba(0, 0, 0, 0.1), inset 0 1px 5px rgba(0, 0, 0, 0.06);
```
Comment (dmcalester) [inline src/css/common/_theme--light.css:161]: remove

File: src/css/common/_theme--light.css [hunk 16]
```diff
@@ -149,19 +149,19 @@
...
+  	transition: box-shadow 0.067s ease-out; */
+	--segmentedButtonSelectedBoxShadow: none;
```
Comment (dmcalester) [inline src/css/common/_theme--light.css:163]: remove

File: src/css/common/_theme--dark.css [hunk 17]
```diff
@@ -131,22 +131,23 @@
...
+  --segmentedButtonTextColor: var(--primary);
+
+	/* Segmented Hover */
+  --segmentedButtonHoverBackgroundColor: var(--primaryLight);
+  --segmentedButtonHoverTextColor: var(--primaryElementText);
+
+	/* Segmented Select */
+  --segmentedButtonSelectedBackgroundColor: var(--primary);
+
+	/* --segmentedButtonSelectedBoxShadow:
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:145]: remove

File: src/css/common/_theme--dark.css [hunk 18]
```diff
@@ -131,22 +131,23 @@
...
+  --segmentedButtonHoverTextColor: var(--primaryElementText);
+
+	/* Segmented Select */
+  --segmentedButtonSelectedBackgroundColor: var(--primary);
+
+	/* --segmentedButtonSelectedBoxShadow: 
+		inset 0 2px 4px rgba(0, 0, 0, 0.14),
+		inset 0 3px 4px rgba(0, 0, 0, 0.12),
+		inset 0 1px 5px rgba(0, 0, 0, 0.2); */
+		--segmentedButtonSelectedBoxShadow:none;
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:149]: remove

**Review by dmcalester — CHANGES_REQUESTED** [review]
Minor remove box-shadow vars, props and associated transitions

**Comment by im-cr** [comment]
@dmcalester box shadows and transitions removed

**Comment by amalyah** [comment]
Missing hover state on selected button section. @carmenparks Is this okay or should we add in a hover state for the selected segment as well?

**Comment by amalyah** [comment]
Hover state still uses the primary color for the border line around the hover fill colors. Is this acceptable @carmenparks? The design file has the same border color as the background color on hovered segments. We would have to alter the design to align with the code.

**Review by amalyah — CHANGES_REQUESTED** [review]
Need alignment with designs or approval from Visual Design to alter the designs to match the code.

**Comment by amalyah** [comment]
Font size needs to change to 16px equivalent (A5 KM v1.1 body font size)

**Comment by carmenparks** [comment]
> Missing hover state on selected button section. @carmenparks Is this okay or should we add in a hover state for the selected segment as well?

Selected state should not have a hover since no action can be taken when it's selected. It's fine as is.

**Comment by carmenparks** [comment]
> Hover state still uses the primary color for the border line around the hover fill colors. Is this acceptable @carmenparks? The design file has the same border color as the background color on hovered segments. We would have to alter the design to align with the code.

Hover state border should match the fill.

**Review by amalyah — CHANGES_REQUESTED** [review]
Needs font size change to body font size. Also, hover state should have border color of that segment be the same as the background color of the hovered segment.

**Comment by im-cr** [comment]
> Needs font size change to body font size. Also, hover state should have border color of that segment be the same as the background color of the hovered segment.

font-size and hove state border updated

**Review by amalyah — APPROVED** [review]

**Review by cskerbo — APPROVED** [review]

**Pull request merged**